docs/Toy/toy/rewrites/inline_toy.py

```diff
@@ -0,0 +1,102 @@
+from xdsl.ir import MLContext
+from xdsl.dialects.builtin import (
+    ModuleOp,
+    StringAttr,
+)
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriteWalker,
+    op_type_rewrite_pattern,
+    RewritePattern,
+    PatternRewriter,
+)
+from xdsl.transforms.dead_code_elimination import dce
+
+from ..dialects import toy
+
+
+class InlineFunctions(RewritePattern):
+    _func_op_by_name: dict[str, toy.FuncOp] | None = None
+
+    def func_op(self, module: ModuleOp, name: str) -> toy.FuncOp:
```

Comment:
You might want to have a separate pass to collect all of these functions in one go, rather than looking it up at each iteration. Not necessarily though, it's up to you.

Reply:
That's what I do below